src/components/celestialmap/TheSkyChart.vue

```diff
@@ -1,431 +1,234 @@
-                    }
-                });
-                // This retains the crosshairs when redrawing with different filters.
-                // However, crosshairs are updated by themselves so an entire repaint is avoided.
-                Celestial.setStyle(styles.crosshair);
-                var p_coords = ([helpers.hour2degree(that.selectedRa), that.selectedDec]);
-                var t_coords = ([helpers.hour2degree(that.ra), that.dec]);
-                // Limit selectable regions to coordinates on the map with Celestial.clip().
-                if (Celestial.clip(p_coords)) {
-                    //Celestial.Canvas.symbol()
-                    //.type("marker")
-                    //.size(420)
-                    //.position(Celestial.mapProjection(p_coords))(Celestial.context);
-
-                    p_coords = Celestial.mapProjection(p_coords);
-                    var r = 18;
-                    var r2 = 4;
-                    Celestial.context.beginPath();
-                    Celestial.context.moveTo(p_coords[0], p_coords[1] - r);
-                    Celestial.context.lineTo(p_coords[0], p_coords[1] - r2);
-                    Celestial.context.moveTo(p_coords[0], p_coords[1] + r);
-                    Celestial.context.lineTo(p_coords[0], p_coords[1] + r2);
-                    Celestial.context.moveTo(p_coords[0] - r, p_coords[1]);
-                    Celestial.context.lineTo(p_coords[0] - r2, p_coords[1]);
-                    Celestial.context.moveTo(p_coords[0] + r, p_coords[1]);
-                    Celestial.context.lineTo(p_coords[0] + r2, p_coords[1]);
-                    Celestial.context.closePath();
-                    Celestial.context.stroke();
-                }
-                Celestial.setStyle(styles.telescope_crosshair);
-                Celestial.Canvas.symbol()
-                    .type("cross-circle")
-                    .size(220)
-                    .position(Celestial.mapProjection(t_coords))(Celestial.context);
-                Celestial.context.stroke();
-            }});
-        },
 
-        update_pointer() {
-            this.initializePointer();
-            Celestial.redraw();
-        },
+        // Update the center of the map every minute
+        this.updateMapCenterInterval = setInterval(this.rotate, 6000)
+    },
 
-        redraw() {
-            Celestial.redraw()
+    beforeDestroy() {
+        this.resize_observer.disconnect()
+        clearInterval(this.updateMapCenterInterval)
+    },
+
+    methods: {
+
+        /** The following few handler methods deal with data coming from the interaction layer */
+        handle_mousedown(e) {
+            let map_coords = Celestial.mapProjection.invert(e)
+
+            // If the click fell within the visible horizon (not outside the map)
+            if (Celestial.clip(map_coords)) { 
+                this.user_crosshairs = this.pix_to_relative(e) // update the visible interaction layer
+                // Save the selected ra/dec
```

Comment:
This is something I didn't even notice on the site - handy when switching back and forth between tabs, but also back and forth between telescopes if you wanted to point multiple at the same place!